backend/src/data/repositories/workspaceRepository.ts

```diff
@@ -0,0 +1,18 @@
+import { EntityRepository, Repository } from 'typeorm';
+import { Workspace } from '../entities/Workspace';
+import { ICreateWorkspace, IWorkspaceResponse } from '../../common/models/workspace';
+
+@EntityRepository(Workspace)
+export class WorkspaceRepository extends Repository<Workspace> {
+  async addWorkspace(data: ICreateWorkspace): Promise<IWorkspaceResponse> {
+    const workspace = this.create(data);
+
+    await workspace.save();
+    return workspace;
```

Comment:
workspace has Workspace type, not IWorkspaceResponse. You should map it in service